src/OneDrive.js

```diff
@@ -243,32 +259,40 @@ class OneDrive extends EventEmitter {
 
   /**
    */
-  async getClient(raw = false) {
+  async doFetch(relUrl, rawResponseBody = false, options = {}) {
+    const opts = { ...options };
     const { accessToken } = await this.getAccessToken();
-    const opts = {
-      baseUrl: 'https://graph.microsoft.com/v1.0',
-      json: true,
-      auth: {
-        bearer: accessToken,
-      },
-    };
-    if (raw) {
-      delete opts.json;
-      opts.encoding = null;
+    if (!opts.headers) {
+      opts.headers = {};
     }
-    return rp.defaults(opts);
-  }
-
-  async me() {
+    opts.headers.authorization = `Bearer ${accessToken}`;
+    const url = `https://graph.microsoft.com/v1.0${relUrl}`;
     try {
-      return await (await this.getClient()).get('/me');
+      const resp = await fetch(url, opts);
+      if (!resp.ok) {
+        const err = StatusCodeError.fromErrorResponse(await resp.json(), resp.status);
+        if (err.statusCode === 404) {
+          this.log.warn(`${relUrl} : ${err.statusCode} - ${err.message}`);
+        } else {
+          this.log.error(`${relUrl} : ${err.statusCode} - ${err.message} - ${err.details}`);
+        }
+        throw err;
+      }
+      return rawResponseBody ? resp.buffer() : resp.json();
```

Comment:
should we `await` here to catch the error?

Reply:
done